### PR TITLE
Feat: Fetched Trending and New Posts in admin dashboard successfully Issue 366

### LINKED ISSUE #366: [Feature Request]: Fetching Trending and New Posts in admin dashboard
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Feature Description

I will be fetching Trending and New Posts in admin dashboard.
Also i will limit the posts table till 6 posts only so that Ui will maintained througout the dashboard.

### Use Case

Admin Visualization

### Benefits

Up to date

### Add ScreenShots

image

### Priority

High

### Record

- [X] I have read the Contributing Guidelines
- [X] I'm a GSSOC'24 contributor
- [X] I have starred the repository

### PR DESCRIPTION
**Pull request by MeetDOD**
# Pull Request Resolves [#366 ]

### Description

1. I am fetching the posts of total 6 in both the table, fetching from posts with pagination and the trending posts.
2. Created interface for post and user.

### Related Issues
Fixes #366 

### Changes Made
New features: Fetched Trending and New Posts in admin dashboard.

### Screenshots

image

image

I certify that I have carried out the relevant code of conduct and provided the requisite screenshot for validation by submitting this pull request.

Thank You for this contribution.

### CODE REVIEW

**Comment by MeetDOD** [comment]
@VaibhavArora314 @Ultimateutkarsh11 @akbatra567 please review this 😄

**Review by VaibhavArora314 — CHANGES_REQUESTED** [review]
I think we should not provide fields such as email on the posts endpoint since email has to be hidden, rather you can create a separate endpoint for admin/post to get posts with email and other extra details.

**Comment by MeetDOD** [comment]
> I think we should not provide fields such as email on the posts endpoint since email has to be hidden, rather you can create a separate endpoint for admin/post to get posts with email and other extra details.

I didn't get it what are you saying, can explain me in brief please :_)

**Comment by MeetDOD** [comment]
> I think we should not provide fields such as email on the posts endpoint since email has to be hidden, rather you can create a separate endpoint for admin/post to get posts with email and other extra details.

Like even if we hide the email directly or indirectly anyone can see the email ! no hiding is not the solution i think .

**Comment by MeetDOD** [comment]
@VaibhavArora314 Done the changes as requested level 3 because of api and frontend 😄 !

File: admin/src/components/NewPosts.tsx [hunk 1]
```diff
@@ -1,6 +1,31 @@
...
+
+  const fetchPost = async () => {
+    try {
+      const response = await axios.get('/api/v1/admin/getAdminAllPosts',{
+        headers: {
+          Authorization: `Bearer ${token}`,
+        },
+      });
+      setposts(response.data.posts.reverse());
+      console.log(response.data.posts);
```
Comment (akbatra567) [inline admin/src/components/NewPosts.tsx:20]: remove this console.log statement.

File: admin/src/components/TrendingPosts.tsx [hunk 2]
```diff
@@ -1,96 +1,79 @@
...
+
+  const fetchPost = async () => {
+    try {
+      const response = await axios.get('/api/v1/admin/getAdminAllTrendingPosts',{
+        headers: {
+          Authorization: `Bearer ${token}`,
+        },
+      });
+      setTrendingPosts(response.data.trendingPosts);
+      // console.log(response.data.trendingPosts);
```
Comment (akbatra567) [inline admin/src/components/TrendingPosts.tsx:20]: remove the comment.

**Review by VaibhavArora314 — CHANGES_REQUESTED** [review]
change route names to /admin/posts/all and /admin/posts/trending

**Comment by MeetDOD** [comment]
> change route names to /admin/posts/all and /admin/posts/trending

@VaibhavArora314 Changed the routes as per request 👍

File: admin/src/components/NewPosts.tsx (hunk 1 above)
Comment (MeetDOD) [reply admin/src/components/NewPosts.tsx]: Removed

File: admin/src/components/TrendingPosts.tsx (hunk 2 above)
Comment (MeetDOD) [reply admin/src/components/TrendingPosts.tsx]: Removed

**Review by VaibhavArora314 — APPROVED** [review]

**Review by akbatra567 — APPROVED** [review]

**Review by Ultimateutkarsh11 — COMMENTED** [review]
@MeetDOD kindly resolve the merge conflicts

**Comment by MeetDOD** [comment]
> @MeetDOD kindly resolve the merge conflicts

Done please add level 3 label because of api + frontend 😄

**Review by Ultimateutkarsh11 — APPROVED** [review]

**Comment by MeetDOD** [comment]
@Ultimateutkarsh11 can i work on issue #389 now because the block feature is done ?

**Comment by Ultimateutkarsh11** [comment]
> @Ultimateutkarsh11 can i work on issue #389 now because the block feature is done ?

Yes, go ahead.

**Pull request merged**